src/src/filemanager/hillel/FileNavigator.java

```diff
@@ -0,0 +1,65 @@
+package filemanager.hillel;
+
+import java.util.*;
+
+public class FileNavigator {
+
+    private Map<String, List<FileData>> fileMap;
+
+    public FileNavigator() {
+        fileMap = new HashMap<>();
+    }
+
+    public void add(String path, FileData file) {
+        if (path.equals(file.getPath())) {
+            List<FileData> fileList = fileMap.get(path);
+            if (fileList == null) {
+
+                fileList = new ArrayList<>();
+                fileMap.put(path, fileList);
+            }
+            fileList.add(file);
+        } else {
+            throw new RuntimeException("Different path: " + path +" "+file.getPath());
+        }
+    }
+
+    public List<FileData> find(String path) {
+        List<FileData> fileList = fileMap.get(path);
+        if (fileList == null) {
+            System.out.println("No files found at path " + path);
+            return Collections.emptyList();
+        }
+        return fileList;
+    }
+
+    public List<FileData> filterBySize(String path, int maxSize) {
```

Comment:
you do not need a path here, only a size. You should filter among all files in a map.